### DISCUSSION TITLE
Chakra version 3 using IntelliJ/WebStorm, some components have no code completion

### DISCUSSION THREAD

**Discussion by andreasnaess** [chakra-ui/chakra-ui#9159] [Q&A] [answered]
I tried to create a new Vite project with Chakra version 3, but I encountered an error where IntelliJ/WebStorm is not showing code completion for some components.

For example: the Flex and Button components get no code completions, but the Box and Center does.
The code completion issue does not happen with Chakra version 2 for these components.

Am I missing something?

I have the following package.json:

```
{
  "name": "demo-project",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc -b && vite build",
    "lint": "eslint .",
    "preview": "vite preview"
  },
  "dependencies": {
    "@chakra-ui/react": "^3.1.1",
    "@emotion/react": "^11.13.3",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@eslint/js": "^9.13.0",
    "@types/react": "^18.3.12",
    "@types/react-dom": "^18.3.1",
    "@vitejs/plugin-react": "^4.3.3",
    "eslint": "^9.13.0",
    "eslint-plugin-react-hooks": "^5.0.0",
    "eslint-plugin-react-refresh": "^0.4.14",
    "globals": "^15.11.0",
    "typescript": "~5.6.2",
    "typescript-eslint": "^8.11.0",
    "vite": "^5.4.10"
  }
}
```

**Comment by Kanat1ch** [comment]
The same problem... Did you manage to solve it?

**Comment by andreasnaess** [comment]
I have not solved the issue. It appears to be related to component types: components of type ChakraComponent work as expected, whereas those of type ForwardRefExoticComponent do not.

**Comment by Davasny** [comment, marked as answer]
I found the solution, in webstorm we have to check "use types from server"

<img width="1011" alt="image" src="https://github.com/user-attachments/assets/b9b6997b-7515-47ac-8d16-1f8c8053229a" />

<img width="1428" alt="image" src="https://github.com/user-attachments/assets/f9152b19-bc5e-4bb0-862e-be0a46ed7990" />
